src/azure-cli-core/azure/cli/core/local_context.py

```diff
@@ -0,0 +1,124 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import stat
+import shutil
+import configparser
+
+from knack.log import get_logger
+from knack.config import _ConfigFile
+from knack.util import ensure_dir, CLIError
+
+STORE = 'STORE'   # action for a parameter in local context, STORE means its value will be saved to local context
+USE = 'USE'   # action for a parameter in local context, USE means will read value from local context for this parameter
+ALL = 'ALL'   # effective level of local context, ALL means all commands can share this parameter value
+logger = get_logger(__name__)
+
+
+class AzCLILocalContext(object):
+
+    def __init__(self, dir_name=None, file_name=None):
+        self.dir_name = dir_name
+        self.file_name = file_name
+        self._local_context_file = None
+        self._load_local_context_file()
+
+    def _load_local_context_file(self):
+        current_dir = os.getcwd()
+        while current_dir:
+            dir_path = os.path.join(current_dir, self.dir_name)
+            file_path = os.path.join(dir_path, self.file_name)
+            if os.path.isfile(file_path) and os.access(file_path, os.R_OK) and os.access(file_path, os.W_OK):
+                self._local_context_file = _ConfigFile(dir_path, file_path)
+                break   # load only one local context
+            # Stop if already in root drive
+            if current_dir == os.path.dirname(current_dir):
+                if self._local_context_file is None:
+                    logger.debug('local context is not turned on in %s and all its parent directories', current_dir)
```

Comment:
current_dir is now root, should it be the original directory get by os.getcwd()?

Reply:
this is just debug information, for simplicity, let's just keep this message.